samples/tutorials/Data passing in python components.ipynb

```diff
@@ -0,0 +1,305 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Data passing tutorial\n",
+    "This tutorial shows how to create python components that produce, consume and transform data.\n",
+    "It shows how to create data passing pipelines by instantiating components and connecting them together."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from typing import NamedTuple\n",
+    "\n",
+    "import kfp\n",
+    "from kfp.components import InputPath, InputTextFile, InputBinaryFile, OutputPath, OutputTextFile, OutputBinaryFile\n",
+    "from kfp.components import func_to_container_op"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Small data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Consuming small data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@func_to_container_op\n",
+    "def consume_one_argument(text: str):\n",
+    "    '''Print small text'''\n",
+    "    print(text)\n",
+    "\n",
+    "def constant_to_consumer_pipeline():\n",
+    "    '''Pipeline that passes small constant string to to consumer'''\n",
+    "    consume_task = consume_one_argument('Hello world') # Passing constant as argument to consumer\n",
+    "\n",
+    "kfp.Client().create_run_from_pipeline_func(constant_to_consumer_pipeline, arguments={})"
```

Comment:
I'm hesitant on introducing additional ways to build / push pipelines specifically referring to create_run_from_pipeline_func() . I understand that this method was not introduced in this example however since it will be highlighted in this example we may need to revisit it. @gaoning777 @hongye-sun @jessiezcc

Reply:
I think that the tutorials are easier for the users to understand if it does not contain too much boilerplate. Adding boilerplate will make the sample almost twice as big and make it quite unreadable.
Since this tutorial is geared towards the people who just start I really do not want to introduce many complicated concepts that are not relevant to the tutorial (e.g. Compilation, pipeline packages, Experiments, Runs) and all their intricate details that can make the user experience fail (e.g. failures when re-running the same notebook due to run name being the same, etc).

If the user wants to learn the more advanced concepts later, they can always do that. But I'm not sure we should set the bar too high for this entry-level tutorial.